hack/ci/run-argocd-e2e-tests.sh

```diff
@@ -0,0 +1,313 @@
+    tofu output -json > tf.json && ../../../${KUBEONE_INSTALL_DIR}/kubeone apply -t ./tf.json -m kubeone.yaml --auto-approve
+  if [ $? -ne 0 ]; then
+    echodate kubeone master cluster installation failed.
+    exit 2
+  fi
+  cd ../..
+  aws s3 cp ${MASTER_KUBECONFIG} s3://cluster-backup-e2e/kkp-argocd-test/kubeconfig/
+
+  if [[ ${SEED} != false ]]; then
+    cd kubeone-install/${SEED} && tofu init && tofu apply -auto-approve &&
+      tofu output -json > tf.json && ../../../${KUBEONE_INSTALL_DIR}/kubeone apply -t ./tf.json -m kubeone.yaml --auto-approve
+    # cd kubeone-install/${SEED} && tofu init && tofu apply -auto-approve
+    if [ $? -ne 0 ]; then
+      echodate kubeone seed cluster installation failed.
+      exit 3
+    fi
+    cd ../..
+    aws s3 cp ${SEED_KUBECONFIG} s3://cluster-backup-e2e/kkp-argocd-test/kubeconfig/
+  fi
+}
+
+# Validate kubeone clusters - apiserver availability, smoke test
+# TODO: do via chainsaw as well as check apiserver availability
+validateSeedClusters() {
+  echodate validateSeedClusters: Not implemented.
+}
+
+# deploy argo and kkp argo apps
+deployArgoApps() {
+  echodate Deploying ArgoCD and KKP ArgoCD Apps.
+  # TODO: variable for the ingress hostname
+  helm repo add dharapvj https://dharapvj.github.io/helm-charts/
+  helm repo add argo https://argoproj.github.io/argo-helm
+  helm repo update dharapvj
+  helm repo update argo
+  # master seed
+  KUBECONFIG=${MASTER_KUBECONFIG} helm upgrade --install argocd --version ${ARGO_VERSION} --namespace argocd --create-namespace argo/argo-cd -f values-argocd.yaml --set "server.ingress.hosts[0]=argocd.${CLUSTER_PREFIX}.lab.kubermatic.io" --set "server.ingress.tls[0].hosts[0]=argocd.${CLUSTER_PREFIX}.lab.kubermatic.io"
+  KUBECONFIG=${MASTER_KUBECONFIG} helm upgrade --install kkp-argo-apps --set kkpVersion=${KKP_VERSION} -f ./${ENV}/demo-master/argoapps-values.yaml dharapvj/argocd-apps
+
+  if [[ ${SEED} != false ]]; then
+    KUBECONFIG=${SEED_KUBECONFIG} helm upgrade --install argocd --version ${ARGO_VERSION} --namespace argocd --create-namespace argo/argo-cd -f values-argocd.yaml --set "server.ingress.hosts[0]=argocd.india.${CLUSTER_PREFIX}.lab.kubermatic.io" --set "server.ingress.tls[0].hosts[0]=argocd.india.${CLUSTER_PREFIX}.lab.kubermatic.io"
+    KUBECONFIG=${SEED_KUBECONFIG} helm upgrade --install kkp-argo-apps --set kkpVersion=${KKP_VERSION} -f ./${ENV}/india-seed/argoapps-values.yaml dharapvj/argocd-apps
+  fi
+}
+# download kkp release and run kkp installer
+installKKP() {
+  echodate installing KKP on master seed.
+  if [ ! -d "${INSTALL_DIR}" ]; then
+    echodate "$INSTALL_DIR does not exist. Downloading KKP release"
+    BIN_ARCH=linux-amd64
+    mkdir -p ${INSTALL_DIR}/
+    curl -sL "https://github.com/kubermatic/kubermatic/releases/download/${KKP_VERSION}/kubermatic-ee-${KKP_VERSION}-${BIN_ARCH}.tar.gz" | tar -xz --directory ${INSTALL_DIR}/
+  fi
+
+  # replace imagepullsecret
+  export DECODE=$(echo $IMAGE_PULL_SECRET_DATA | base64 -d)
```

Comment:
Will this work if the image pull secret is a multiline JSON string (after decoding, I mean)? Should this be quoted?

Reply:
pull-secret that is getting pulled from vault is already multiline json string. This above part works fine. `yq` is intelligent about using quotes when needed.